### ISSUE TITLE
yum module can't add extra repo entry to /etc/yum.repos.d

### ISSUE THREAD

**Issue by ghost** [ansible/ansible#11814]
**Issue Type:** 
Bug Report

**Ansible Version:**
Running from source (cloned from GitHub)

``` text
ansible 2.0.0 (devel b37be236d9) last updated 2015/07/30 15:55:40 (GMT +800)
  lib/ansible/modules/core: (detached HEAD a46b3a4dc4) last updated 2015/07/30 16:03:22 (GMT +800)
  lib/ansible/modules/extras: (detached HEAD e30d8b84fe) last updated 2015/07/30 16:10:42 (GMT +800)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/07/30 16:21:25 (GMT +800)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/07/30 16:21:35 (GMT +800)
  config file = 
  configured module search path = None
```

**Ansible Configuration:**
Just the original configuration, nothing changed.

**Environment:**
**_Control Host**_: Ubuntu 14.04.1 (kernel: 3.16.0-30-generic)
**_Target Host**_: CentOS 7 (kernel: 3.10.0-123.e17.x86_64)

**Summary:**
A extra repo entry (_erlang-solutions.repo_) will be added to yum repo folder (`/etc/yum.repos.d/`), when I install _"erlang-solutions-1.0-1.noarch.rpm"_ **directly on the host** using the following commands:

``` bash
wget http://packages.erlang-solutions.com/erlang-solutions-1.0-1.noarch.rpm
rpm -Uvh erlang-solutions-1.0-1.noarch.rpm
```

**_BUT!!!!!**_ After installed _"erlang-solutions-1.0-1.noarch.rpm"_ through **Ansible yum module**, there's no extra repo entry (_erlang-solutions.repo_) appeared in yum repo folder (`/etc/yum.repos.d/`). 

**Steps To Reproduce:**

``` yaml
- name: install erlang-solutions
  yum: name=http://package.erlang-solutions.com/erlang-solutions-1.0-1.noarch.rpm state=present
```

**Expected Results:**
There's a extra repo entry (_erlang-solutions.repo_) added in yum repo folder (`/etc/yum.repos.d/`)

**Actual Results:**
The package (_erlang-solutions-1.0-1_) was **installed successfully**, **BUT** the repo entry (_erlang-solutions.repo_) was **NOT** added.

**Comment by abadger** [comment]
Is your playbook example an exact cut and paste of your actual playbook?  If so, the reason this is happening is probably a typo:  http://package.erlang-solutions.com should be http://packages.erlang-solutions.com

I'm checking what "yum install" does on centos7 now.  If that's the problem it feels like we should return an error message but if yum itself doesn't give us an error here then it might not be something that we can support.

**Comment by abadger** [comment]
I get an error message:

```
$ ansible-playbook --sudo -K test.yml
SUDO password: 

PLAY ****************************************************************************

TASK [setup] ********************************************************************
ok: [localhost]

TASK [install erlang-solutions] *************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Package at http://package.erlang-solutions.com/erlang-solutions-1.0-1.noarch.rpm could not be installed", "rc": 1}

PLAY RECAP **********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1   
```

Checking what the package at the proper URL looks like now.

**Comment by abadger** [comment]
Looking at the actual package in case that's the problem, it looks like the rpm package from erlang-solutions is doing some really hacky things that we can't fix.  the package doesn't include a repo file.  Instead, it uses an rpm %post scriptlet to wget a repo file from erlang-solutions.com.  This is very fragile and that fragility is probably what's happening here.

If the %post script fails (on centos7 at least), yum decides it's nonfatal to the package installation and therefore it exits without an error code.  So as far as ansible can tell, the package installed correctly.  However, the repo file isn't installed because the repo file isn't part of the package; it's a side effect of the package being installed.

I was able to make that result in no repo file install by not having wget installed on the target system.  Other things (like network errors) can probably cause the same symptoms.  If everything is correct on the host machine to make sure the package's %post scriptlet runs then the repo file is downloaded fine.

To summarize, there's no bug in ansible here.  It's a bad rpm package.  You should talk to the erlang-solutions people about creating a better rpm package.  They probably want to have something more like this in their rpm spec file:

```
Source0: http://packages.erlang-solutions.com/rpm/fedora/erlang_solutions.repo
Source1: http://packages.erlang-solutions.com/rpm/centos/erlang_solutions.repo
[...]
%install
%if 0%{?fedora} 
cp %{SOURCE0} %{buildroot}/etc/yum.repos.d/
%endif

%if 0%{?rhel}
cp %{SOURCE1} %{buildroot}/etc/yum.repos.d/
[...]
%files
%{_sysconfdir}/yum.repos.d/*
```

**Comment by abadger** [comment]
# Is This A Bug?

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We're not sure this is a bug, and we don't mean for this to be confrontational.  Let's explain what we're thinking:
- The rpm in question is attempting to install the repo file via a %post script.  That is a fragile way to do this and leads to yum reporting successful installation of the package (which is true) but the post install script hasn't completed successfully and therefore the repo file has not been downloaded on the system.  You should talk to the erlang-solutions people about creating a better rpm that won't have this problem.

As such, we're going to close this ticket.  However, we're open to being corrected, should you wish to discuss.  You can stop by one of our two mailing lists 
to talk about this and we might be persuaded otherwise.
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Comments on closed tickets aren't something we monitor, so if you do disagree with this, a mailing list thread is probably appropriate.

Thank you once again for this and your interest in Ansible!

**Comment by ghost** [comment]
Sorry for replying so late! Like you've explained, this might be the problem of the rpm package. Anyway, thanks for your reply!

By the way, Ansible is a great project :+1:
